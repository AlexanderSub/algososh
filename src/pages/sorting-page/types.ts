export interface Step {
  currentArray: number[]
  sortedIndexes: number[]
  aIndex?: number
  bIndex?: number
}

export enum SortType {
  Select = 'select',
  Bubble = 'bubble'
}


