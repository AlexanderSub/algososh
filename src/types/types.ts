import { ElementStates } from "./element-states";

export type TArrayString = {
  value: string;
  state: ElementStates;
}

export type TArrayNumber = {
  value: number;
  state: ElementStates;  
}

export enum Direction {
  Ascending = "ascending",
  Descending = "descending",
}