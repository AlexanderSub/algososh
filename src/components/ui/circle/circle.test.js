import {Circle} from './circle'
import renderer from 'react-test-renderer'
import { ElementStates } from '../../../types/element-states'

describe('Circle', () => {
  it('Пустой круг рендерится без ошибок', () => {
    const tree = renderer.create(
      <Circle/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Круг с буквой рендерится без ошибок', () => {
    const tree = renderer.create(
      <Circle letter='A'/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Круг с head', () => {
    const tree = renderer.create(
      <Circle head={'test'}/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Круг с react-элементом в head', () => {
    const tree = renderer.create(
      <Circle head={<p>test</p>}/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Круг с tail', () => {
    const tree = renderer.create(
      <Circle tail={'test'}/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Круг с react-элементом в tail', () => {
    const tree = renderer.create(
      <Circle tail={<p>test</p>}/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Круг с index', () => {
    const tree = renderer.create(
      <Circle index={1}/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Круг c пропом isSmall ===  true', () => {
    const tree = renderer.create(
      <Circle isSmall/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Круг в состоянии default', () => {
    const tree = renderer.create(
      <Circle state={ElementStates.Default}/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Круг в состоянии changing', () => {
    const tree = renderer.create(
      <Circle state={ElementStates.Changing}/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Круг в состоянии modified', () => {
    const tree = renderer.create(
      <Circle state={ElementStates.Modified}/>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})