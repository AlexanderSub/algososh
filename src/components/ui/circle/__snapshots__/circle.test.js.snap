// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Круг c пропом isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Круг в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Круг в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Круг в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Круг с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    test
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Круг с index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle Круг с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <p>
      test
    </p>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Круг с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <p>
      test
    </p>
  </div>
</div>
`;

exports[`Circle Круг с tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    test
  </div>
</div>
`;

exports[`Circle Круг с буквой рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle Пустой круг рендерится без ошибок 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
